### ISSUE TITLE
senec-collector hang up

### ISSUE THREAD

**Issue by iot-sle** [solectrus/senec-collector#158]
Tonight the first time the senec-collector stopped working. (Synology / Docker) Solectrus 0.7.7 
The container was still up and running but no values are send to the InfluxDB any longer. After restart of the collector container everything was fine again. The log of the container didn't show entries.

```
2023-02-03T06:32:03.781560362Z,stdout,Successfully pushed record to InfluxDB
2023-02-03T06:32:02.700521174Z,stdout,"Got record #1 from SENEC at 192.168.252.60: Inverter 0 W, House 

2023-02-03T06:32:01.287215096Z,stdout,"Pushing to InfluxDB at http://influxdb:8086, bucket my-solectrus-bucket"
2023-02-03T06:32:01.287178405Z,stdout,Pulling from SENEC at 192.168.252.60 every 5 seconds

2023-02-03T06:32:01.287142777Z,stdout,Using Ruby 3.1.3 on platform x86_64-linux-musl
2023-02-03T06:32:01.287066316Z,stdout,"Copyright (c) 2020,2022 Georg Ledermann, released under the MIT License"
2023-02-03T06:32:01.286986197Z,stdout,https://github.com/solectrus/senec-collector
2023-02-03T06:32:01.286459460Z,stdout,SENEC collector for SOLECTRUS
2023-02-03T00:18:05.534552184Z,stdout,Successfully pushed record to InfluxDB

2023-02-03T00:18:05.420491819Z,stdout,"Got record #173942 from SENEC at 192.168.252.60: Inverter 0 W, House 255 W, 2023-02-03 00:18:06 +0000"

2023-02-03T00:18:00.391848261Z,stdout,"Got record #173941 from SENEC at 192.168.252.60: Inverter 0 W, House 256 W, 2023-02-03 00:18:01 +0000"

```

**Comment by ledermann** [comment]
Hm, interesting, I have not seen something like that with my installation. 

You have collected 173942 records without restarting the container. Based on the 5-second interval, that's `173942 * 5 / 60 / 60 / 24 = 10` days. That is not so much, IMHO. I also use a Synology, but the container is restarted more often because the image is updated regularly (using Watchtower). 

Maybe a memory leak? I will take a closer look at this.

**Comment by iot-sle** [comment]
you're right. I only restart the container when an update is available. I do it manually by using the update hint at the solectrus version showing on the web page.

**Comment by ledermann** [comment]
Note that the Dashboard app (`ghcr.io/solectrus/solectrus`) and the senec-collector (`ghcr.io/solectrus/senec-collector`) are different Docker images which are released and updated separately. The update notification in the Dashboard UI does **not** check the senec-collector.

Back to the problem: There appears to be a memory leak, as the memory consumption is continuously increasing over time (according to `docker stats`). However, I have not found the cause yet. I will continue to keep an eye on this.

**Comment by alexdetsch** [comment]
Can confirm the issue in the current version 0.6.0. Had recorded from 2023-02-12T11:52:00 until 2023-02-21T21:43:00 in a 2 second interval resulting in 233235 records until it hang up without notice

**Comment by ledermann** [comment]
Current state: There is hope :-)

Today I rewrote large parts of the collector (two threads communicating via a queue, great fun!) - and it seems to me that the memory leak is fixed. I will monitor this for a few days before releasing a new version.

**Comment by ledermann** [comment]
Just released `v0.7.0` of the senec-collector, which now no longer consumes all the memory. After two weeks on my Raspberry, the memory consumption is still below 30MB (reported by `docker stats`).

This should be worth an upgrade, @iot-sle and @alexdetsch!

**Comment by ledermann** [comment]
Seems to be fixed successfully. Closing.
